### PR TITLE
[SDK Migration] Switch to celestia-da, bump rollkit, fix CI

### PR DESCRIPTION
**Pull request by okdas**
- Updates rollkit to the newer version;
- Upgrades Golang for CI and local dev tooling (neeed for ignite);
- Bumps ignite version on CI so we can get them sweet green CI runs;
- Bumps `local-celestia-devnet` container that has new [celestia-da](https://github.com/rollkit/celestia-da) by rollkit team. https://github.com/rollkit/local-celestia-devnet/pkgs/container/local-celestia-devnet container exposes port  `26650` for us to access this service. New Rollkit requires celestia-da.

### CODE REVIEW

**Comment by okdas** [comment]
Related helm chart change: https://github.com/pokt-network/helm-charts/pull/27

File: go.mod [hunk 1]
```diff
@@ -336,4 +337,4 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace github.com/cosmos/cosmos-sdk => github.com/bryanchriswhite/cosmos-sdk v0.50.1-fix-sequencer-check
```
Comment (okdas) [inline go.mod:339]: dis is important https://github.com/bryanchriswhite/cosmos-sdk/compare/v0.50.1-fix-sequencer-check...okdas:cosmos-sdk:v0.50.1-fix-sequencer-check-2

File: app/app.go [hunk 2]
```diff
@@ -59,8 +59,8 @@ import (
 )
 
 const (
-	AccountAddressPrefix = "cosmos"
-	Name                 = "poktroll"
+	AccountAddressPrefix = "pokt"
```
Comment (okdas) [inline app/app.go:62]: I've noticed addresses no longer start with `pokt`, but with `cosmos` instead. Unless this is the desired behavior, I'm changing them back.

File: app/app.go [hunk 3]
```diff
@@ -59,8 +59,8 @@ import (
...
+	Name                 = "pocket"
```
Comment (Olshansk) [inline app/app.go:63]: Is this going to change the binary to `pocketd` instead of `poktrolld`. 

It'll impact the documentation if so.

**Review by Olshansk — CHANGES_REQUESTED** [review]

**Comment by Olshansk** [comment]
@okdas Reminder to update metadata.

Screenshot 2024-02-07 at 11 12 05 AM

File: .github/workflows/main-build.yml [hunk 4]
```diff
@@ -29,7 +29,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version: "1.20.10"
+          go-version: "1.21.6"
```
[the remark this answers is not in the document]
Comment (okdas) [reply .github/workflows/main-build.yml:32]: @Olshansk I believe ignite was acting up on 1.20 for me (specifically, it seemed to update `go.mod` to add 1.21 specific attributes 1.20 was not working with). 

@red-0ne @bryanchriswhite do you have any objections against upgrading to 1.21?

File: app/app.go (hunk 3 above)
Comment (okdas) [reply app/app.go:63]: I didn't change anything - these values are what we have in main branch: https://github.com/pokt-network/poktroll/blob/main/app/app.go#L141

File: .github/workflows/main-build.yml [hunk 5]
```diff
@@ -17,8 +17,8 @@ jobs:
         # TODO_TECHDEBT: upgrade to the latest Ignite (the latest at the moment of creating a note is 0.28). Need to downgrade to fix CI pipelines. Might be done in scope of #240.
         run: |
           # curl https://get.ignite.com/cli! | bash
-          wget https://github.com/ignite/cli/releases/download/v0.27.2/ignite_0.27.2_linux_amd64.tar.gz
-          tar -xzf ignite_0.27.2_linux_amd64.tar.gz
+          wget https://github.com/ignite/cli/releases/download/v28.2.0/ignite_28.2.0_linux_amd64.tar.gz
+          tar -xzf ignite_28.2.0_linux_amd64.tar.gz
```
Comment (red-0ne) [inline .github/workflows/main-build.yml:21]: Shouldn't we use `v28.1.0` this seems to be what rollkit are recommending in their docs

https://rollkit.dev/tutorials/gm-world#install-ignite-cli-linux

File: .github/workflows/main-build.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (red-0ne) [reply .github/workflows/main-build.yml:32]: I think 1.21+ is mandatory. At least Rollkit doc is demanding it
image

File: .github/workflows/main-build.yml (hunk 5 above)
Comment (okdas) [reply .github/workflows/main-build.yml:21]: @red-0ne `28.2.0` was just released yesterday, I'm sure Rollkit team just didn't have an opportunity to update the docs. The new version seems to work fine.

File: go.mod [hunk 6]
```diff
@@ -18,7 +18,7 @@ require (
 	cosmossdk.io/depinject v1.0.0-alpha.4
```
[the remark this answers is not in the document]
Comment (okdas) [reply go.mod:18]: We do, I'm surprised it got deleted. It was just regenerated, so I'm committing it back! :)

File: Makefile [hunk 7]
```diff
@@ -164,23 +164,27 @@ localnet_up: ## Starts localnet
...
-	cp ${HOME}/.poktroll/config/genesis.json $(POKTROLLD_HOME)/config/
-	ADDRESS=$$(jq -r '.address' $(POKTROLLD_HOME)/config/priv_validator_key.json); \
-	PUB_KEY=$$(jq -r '.pub_key' $(POKTROLLD_HOME)/config/priv_validator_key.json); \
-	# NB: Currently the stake => power calculation is constant; however, cosmos-sdk \
-	# intends to make this parameterizable in the future. \
-	POWER=$$(yq ".validators[0].bonded" ./config.yml | sed 's,000000upokt,,'); \
-	NAME=$$(yq ".validators[0].name" ./config.yml); \
-	jq --argjson pubKey "$$PUB_KEY" '.consensus["validators"]=[{"address": "'$$ADDRESS'", "pub_key": $$pubKey, "power": "'$$POWER'", "name": "'$$NAME'"}]' $(POKTROLLD_HOME)/config/genesis.json > temp.json && mv temp.json $(POKTROLLD_HOME)/config/genesis.json
+# NB: Currently the stake => power calculation is constant; however, cosmos-sdk
+# intends to make this parameterizable in the future.
```
Comment (okdas) [inline Makefile:172]: For some reason, make quietly ignored everything after the comment section. So I moved them out of code.

**Comment by Olshansk** [comment]
@okdas Re-request review when ready.

File: localnet/poktrolld/config/app.toml [hunk 8]
```diff
@@ -1,264 +1,58 @@
-# This is a TOML config file.
```
[the remark this answers is not in the document]
Comment (okdas) [reply localnet/poktrolld/config/app.toml:1]: @Olshansk this is indeed automatic.

File: app/app.go (hunk 3 above)
Comment (Olshansk) [reply app/app.go:63]: The main branch has `pocket` but this is a PR against `migration/base`, which has `poktroll`.

Is this intended?

cc @bryanchriswhite

**Review by Olshansk — COMMENTED** [review]
Left one comment to understand. Also would like an approval from @bryanchriswhite on this

File: go.mod [hunk 9]
```diff
@@ -336,4 +337,4 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace github.com/cosmos/cosmos-sdk => github.com/bryanchriswhite/cosmos-sdk v0.50.1-fix-sequencer-check
+replace github.com/cosmos/cosmos-sdk => github.com/okdas/cosmos-sdk v0.50.1-fix-sequencer-check-2
```
Comment (bryanchriswhite) [inline go.mod]: As [discussed](https://discord.com/channels/824324475256438814/1138895490331705354/1205219131729776640), we can use theirs now.
```suggestion
replace github.com/cosmos/cosmos-sdk => github.com/rollkit/cosmos-sdk v0.50.1-rollkit-v0.11.19
```

File: docs/static/openapi.yml
Comment (bryanchriswhite) [inline docs/static/openapi.yml:1]: nit: I think this should probably be reverted:
```bash
git checkout migration/base -- docs.static.openapi.yml
```

File: Makefile
Comment (bryanchriswhite) [inline Makefile:1]: As [discussed](https://discord.com/channels/824324475256438814/1138895490331705354/1205220793982656523), we should use `yq` to populate the `$NAME` and `$POWER` variables and interpolate them into the `jq` arg string.

**Review by bryanchriswhite — APPROVED** [review]
Very nice! :sunglasses::ok_hand: 

Seems to be working (at 26c7630fc25aefec92165858288afa5540bfac12): [#365 - #1300](https://github.com/pokt-network/poktroll/actions/runs/7834142601/job/21376710961?pr=365).

Preemptively approving.

File: app/app.go (hunk 3 above)
Comment (bryanchriswhite) [reply app/app.go:63]: I think we actually do want to keep `Name`  as `poktroll`. I'm noticing that it doesn't seem to have an effect on the binary path `.../poktrolld`, however it does have an impact on the `--help` text as well as the default home path and environment variable prefixes from viper:
image

File: Makefile
Comment (bryanchriswhite) [reply Makefile:1]: Ignore me, refresh fail.

File: tools/tools.go [hunk 10]
```diff
@@ -10,6 +10,7 @@ import (
 	_ "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway"
 	_ "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger"
 	_ "github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2"
+	_ "golang.org/x/tools/cmd/goimports"
```
Comment (bryanchriswhite) [inline tools/tools.go:13]: Just out of curiosity, do we know why this was added?

**Comment by red-0ne** [comment]
Experimented with merging this branch with the `Gateway` migration branch and we were able to successfully submit `MsgStakeGateway` transactions :+1:

File: go.mod (hunk 9 above)
Comment (okdas) [reply go.mod]: @bryanchriswhite yup - this branch is on `v0.50.1-rollkit-v0.11.19-no-fraud-proofs`, did you leave the comment on an old commit? 🤔

File: tools/tools.go (hunk 10 above)
Comment (okdas) [reply tools/tools.go:13]: @bryanchriswhite `ignite` (the latest version) prompted to add this. I realized this wouldn't hurt, so I agreed. :) I assume `ignite` uses `goimports` internally.

**Pull request merged**